Sources/MapboxNavigation/NavigationCameraStateTransition.swift

```diff
@@ -167,12 +165,20 @@ public class NavigationCameraStateTransition: CameraStateTransition {
             animatorBearing.stopAnimation()
         }
         
-        let bearingTimingParameters = UICubicTimingParameters(controlPoint1: CGPoint(x: 0.0, y: 0.0), controlPoint2: CGPoint(x: 1.0, y: 1.0))
-        animatorBearing = mapView.camera.makeAnimator(duration: duration, timingParameters: bearingTimingParameters) { (transition) in
-            transition.bearing.toValue = bearing
+        // Check whether the bearing change is larger than a certain threshold when current camera state is following.
+        var updateCameraBearing: Bool = true
+        if state == .following {
+            updateCameraBearing = (abs(mapView.cameraState.bearing - bearing) >= 1.0)
         }
         
-        animatorBearing?.startAnimation()
+        if updateCameraBearing {
+            let bearingTimingParameters = UICubicTimingParameters(controlPoint1: CGPoint(x: 0.0, y: 0.0), controlPoint2: CGPoint(x: 1.0, y: 1.0))
+            animatorBearing = mapView.camera.makeAnimator(duration: duration, timingParameters: bearingTimingParameters) { (transition) in
+                transition.bearing.toValue = bearing
+            }
+            
+            animatorBearing?.startAnimation()
+        }
         
         if let animatorPitch = animatorPitch, animatorPitch.isRunning {
```

Comment:
Should we add a similar check for the pitch? Otherwise, a completely idle map will still be guaranteed to animate. Hopefully a preview/overview map would not want to animate its pitch anyways, so we can probably get away with a literal equality check here.

Reply:
I'll add the check for the pitch change as well to be aligned with the location center and bearing.